pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -243,50 +357,134 @@ func (pl *VolumeZone) Filter(ctx context.Context, cs *framework.CycleState, pod
+	return framework.QueueSkip, nil
+}
+
+// getPVTopologiesFromPVAndAppend append pvTopology to given array and return the array
+func (pl *VolumeZone) getPVTopologiesFromPV(logger klog.Logger, pv *v1.PersistentVolume) []pvTopology {
+	podPVTopologies := make([]pvTopology, 0)
+	for _, key := range topologyLabels {
+		if value, ok := pv.ObjectMeta.Labels[key]; ok {
+			volumeVSet, err := volumehelpers.LabelZonesToSet(value)
+			if err != nil {
+				logger.V(5).Info("Failed to parse label, ignoring the label", "label", fmt.Sprintf("%s:%s", key, value), "err", err)
+				continue
+			}
+			podPVTopologies = append(podPVTopologies, pvTopology{
+				pvName: pv.Name,
+				key:    key,
+				values: sets.Set[string](volumeVSet),
+			})
 		}
-		return framework.AsStatus(err)
 	}
-	return nil
+	return podPVTopologies
 }
 
-// EventsToRegister returns the possible events that may make a Pod
-// failed by this plugin schedulable.
-func (pl *VolumeZone) EventsToRegister() []framework.ClusterEventWithHint {
-	return []framework.ClusterEventWithHint{
-		// New storageClass with bind mode `VolumeBindingWaitForFirstConsumer` will make a pod schedulable.
-		// Due to immutable field `storageClass.volumeBindingMode`, storageClass update events are ignored.
-		{Event: framework.ClusterEvent{Resource: framework.StorageClass, ActionType: framework.Add}},
-		// A new node or updating a node's volume zone labels may make a pod schedulable.
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.UpdateNodeLabel}},
-		// A new pvc may make a pod schedulable.
-		// Due to fields are immutable except `spec.resources`, pvc update events are ignored.
-		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add}},
-		// A new pv or updating a pv's volume zone labels may make a pod schedulable.
-		{Event: framework.ClusterEvent{Resource: framework.PersistentVolume, ActionType: framework.Add | framework.Update}},
+// getPVTopologiesAndSort checks whether target PV is bound to pod's PVC.
+// If PV is bound to pod's PVC, get PVTopologies and Sort.
+func (pl *VolumeZone) getPVTopologiesAndSort(logger klog.Logger, pod *v1.Pod, pv *v1.PersistentVolume) ([]pvTopology, bool) {
+	isPodBoundToPV := false
+	podPVTopologies := make([]pvTopology, 0)
+
+	pvcNames := pl.getPersistentVolumeClaimNameFromPod(pod)
+	for _, pvcName := range pvcNames {
+		pvc, err := pl.pvcLister.PersistentVolumeClaims(pod.Namespace).Get(pvcName)
+		if err != nil {
+			logger.Error(err, "unexpected error during getting PVC", "Namespace", pod.Namespace, "pvc", pvcName)
+			continue
+		}
+
+		// If target PV is bound to pod's PVC, get PVTopologies
+		pvName := pvc.Spec.VolumeName
+		if pvName == pv.Name {
+			isPodBoundToPV = true
+			podPVTopologies = append(podPVTopologies, pl.getPVTopologiesFromPV(logger, pv)...)
+			// One PV is bound to one PVC. One PV can't be bound to more than one PVC.
+			break
```

Comment:
`podPVTopologies` could get only one element, sorting is no longer needed.

Reply:
In my understanding, sorting is necessary.
First, this array flattens if a single PV has multiple labels, so it may have multiple elements. 
Since PersistentVolume labels are editable, if not sorted, it cannot be determined as the same when only the order of the labels of PV changes.

Reply:
> if a single PV has multiple labels, so it may have multiple elements.

Got it, makes sense then.